qa/tasks/cephfs/test_admin.py

```diff
@@ -760,13 +772,42 @@ def test_fs_rename_fails_with_mirroring_enabled(self):
         new_fs_name = 'new_cephfs'
 
         self.run_ceph_cmd(f'fs mirror enable {orig_fs_name}')
+        self.run_ceph_cmd(f'fs fail {self.fs.name}')
         try:
             self.run_ceph_cmd(f'fs rename {orig_fs_name} {new_fs_name} --yes-i-really-mean-it')
         except CommandFailedError as ce:
             self.assertEqual(ce.exitstatus, errno.EPERM, "invalid error code on renaming a mirrored file system")
         else:
             self.fail("expected renaming of a mirrored file system to fail")
         self.run_ceph_cmd(f'fs mirror disable {orig_fs_name}')
+        self.run_ceph_cmd(f'fs set {new_fs_name} joinable true')
+
+    def test_renaming_when_fs_is_not_down(self):
+        '''
+        Test that the command "ceph fs swap" command fails when first of the
+        two of FSs isn't failed/down.
+        '''
+        msg = ('All MDSs for an FS must be down before rename the file '
+               'system. See `ceph fs fail`.')
+        self.negtest_ceph_cmd(
+            args=(f'fs rename {self.fs1.name} --yes_i_really_mean_it'),
+            retval=errno.EINVAL, errmsgs=msg)
```

Comment:
Maybe only check errno? Error string checks can be hard to maintain.

Reply:
I am okay with that. But in lots of other tests we are checking for both.

Reply:
If a check can be based on an errno, we should be using that. If errno isn't available for whatever reason, errno string checks are the next resort.

Reply:
Fixed.